### PR TITLE
[Merged by Bors] - feat: Upper bound on multinomial coefficients with even exponent

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

Co-authored-by: Bhavik Mehta <bhavik.mehta8@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Choose/Multinomial.lean [hunk 1]
```diff
@@ -305,6 +306,25 @@ theorem sum_pow [CommSemiring R] (x : α → R) (n : ℕ) :
 end CommSemiring
 end Finset
 
+namespace Nat
+variable {ι : Type*} {s : Finset ι} {f : ι → ℕ}
+
+lemma multinomial_two_mul_le_mul_multinomial :
+    multinomial s (fun i ↦ 2 * f i) ≤ ((∑ i in s, f i) ^ ∑ i in s, f i) * multinomial s f := by
+  rw [multinomial, multinomial, ← mul_sum]
+  refine Nat.div_le_of_le_mul' ?_
```
Comment (b-mehta) [inline Mathlib/Data/Nat/Choose/Multinomial.lean:315]: Does gcongr make any of this proof cleaner? Or calc-style?

File: Mathlib/Data/Nat/Choose/Multinomial.lean [hunk 2]
```diff
@@ -305,6 +306,25 @@ theorem sum_pow [CommSemiring R] (x : α → R) (n : ℕ) :
...
+  rw [← mul_assoc, ← Nat.mul_div_assoc _ (prod_factorial_dvd_factorial_sum _ _),
+    Nat.le_div_iff_mul_le (by positivity)]
+  refine (Nat.mul_le_mul_right _ $ factorial_two_mul_le _).trans ?_
+  rw [mul_pow, mul_comm, ← mul_assoc, ← mul_assoc]
+  refine Nat.mul_le_mul_right _ (Nat.mul_le_mul_right _ ?_)
+  rw [← Finset.prod_pow_eq_pow_sum, ← prod_mul_distrib]
```
Comment (b-mehta) [inline Mathlib/Data/Nat/Choose/Multinomial.lean:321]:
```suggestion
  rw [← prod_pow_eq_pow_sum, ← prod_mul_distrib]
```

File: Mathlib/Data/Nat/Choose/Multinomial.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Nat/Choose/Multinomial.lean]: Cleaner, not shorter

**Comment by b-mehta** [comment]
bors merge

**Pull request closed without merging**